src/models/calendar-restful-model-collection.ts

```diff
@@ -159,4 +150,32 @@ export default class CalendarRestfulModelCollection extends RestfulModelCollecti
         return Promise.reject(err);
       });
   }
+
+  private buildAvailabilityPayload(
+    options: SingleAvailabilityQuery | ConsecutiveAvailabilityQuery
+  ): Record<string, unknown> {
+    // Instantiate objects from properties to get JSON formatted for the API call
+    const freeBusyJson = options.freeBusy
+      ? options.freeBusy.map(fb => new FreeBusy(fb).toJSON(true))
+      : [];
+    const openHoursJson = options.openHours
+      ? options.openHours.map(oh => new OpenHours(oh).toJSON(true))
+      : [];
+    const calendarsJson = options.calendars
+      ? options.calendars.map(cal => new FreeBusyCalendars(cal).toJSON(true))
+      : [];
+
+    return {
+      emails: options.emails,
+      duration_minutes: options.duration,
+      interval_minutes: options.interval,
+      start_time: options.startTime,
+      end_time: options.endTime,
+      buffer: options.buffer,
+      tentative_busy: options.tentativeBusy,
+      free_busy: freeBusyJson,
+      open_hours: openHoursJson,
+      calendars: calendarsJson,
+    };
```

Comment:
+ `round_robin` and `event_collection_id` only for /availability (not consecutive or free-busy)

Reply:
there will be a few new additions to consecutive soon-ish so we'll keep you in the loop on those

Reply:
scratch that robin bit, just saw that you have it in `src/models/calendar-availability.ts`

Reply:
@tbadun Awesome! I'll add in `event_collection_id`. Just a few questions to make sure I implement it properly:

1. What is the purpose of this field in terms of making an availability call?
2. Is it optional?
3. Is it a string?

Thank you!

Reply:
1. A user can add a collection id to the metadata of one or more events with a given capacity. When looking for availability, the algorithm will grab all events in the specified collection and, if they are not currently at capacity, show those slots as available in addition to the regular availability data. This is mostly used by scheduler.
2. Yes
3. Also yes